### PR TITLE
Added pre-commit hook config

### PR DESCRIPTION
**Pull request by Thulium-Drake**
Hi there,

I added a config file for pre-commit (https://pre-commit.com) to allow users to automatically sign their code when making commits.

I had to make a small edit (of which I'm not sure if it can break anything), if GPG_TTY is already set and ```-p``` is not provided to not change it.

### CODE REVIEW

**Comment by Thulium-Drake** [comment]
From what I've tested so far, the workflow is:

1. Configure the pre-commit
2. Sign your code manually for the first time and add it to the first commit
3. Run the pre-commit hook

There's still one issue I need to resolve, the first invocation of the hook fails, but the second time 'just works (TM)'. I'm not yet sure why

File: src/ansible_sign/cli.py [hunk 1]
```diff
@@ -332,6 +332,8 @@ def gpg_sign(self):
         elif "ANSIBLE_SIGN_GPG_PASSPHRASE" in os.environ:
             self.logger.debug("Taking GPG key passphrase from ANSIBLE_SIGN_GPG_PASSPHRASE env var")
             passphrase = os.environ["ANSIBLE_SIGN_GPG_PASSPHRASE"]
+        elif "GPG_TTY" in os.environ:
```
Comment (shanemcd) [inline src/ansible_sign/cli.py:335]: Can you please pull this out into a separate PR?
